main.js

```diff
@@ -0,0 +1,370 @@
+    if (r) redoStack = JSON.parse(r);
+  }
+
+  loadStacks();
+
+  function setSyncing(on) {
+    pendingRequests += on ? 1 : -1;
+    pendingRequests = Math.max(0, pendingRequests);
+
+    if (pendingRequests > 0) {
+      syncEl.classList.add("syncing");
+      syncEl.textContent = "Syncing...";
+    } else {
+      syncEl.classList.remove("syncing");
+      syncEl.textContent = "Synced ✓";
+    }
+  }
+
+  async function apiFetch(url, options) {
+    setSyncing(true);
+    try {
+      const res = await fetch(url, options);
+      return res;
+    } finally {
+      setSyncing(false);
+    }
+  }
+
+  function clearColumns() {
+    todoCol.innerHTML = "";
+    inprogressCol.innerHTML = "";
+    doneCol.innerHTML = "";
+  }
+
+  function createCard(item) {
+    const card = document.createElement("div");
+    card.className = "task-card";
+    card.setAttribute("draggable", "true");
+    card.dataset.id = item.id;
+
+    const title = document.createElement("div");
+    title.className = "task-title";
+    title.textContent = item.title || item.name || "Untitled";
+
+    const actions = document.createElement("div");
+    actions.className = "task-actions";
+
+    const editBtn = document.createElement("button");
+    editBtn.className = "remove";
+    editBtn.textContent = "✎";
+    editBtn.title = "Edit";
+    editBtn.addEventListener("click", () => editTask(item));
+
+    const delBtn = document.createElement("button");
+    delBtn.className = "remove";
+    delBtn.textContent = "✕";
+    delBtn.title = "Delete";
+    delBtn.addEventListener("click", () => deleteTask(item.id));
+
+    actions.append(editBtn, delBtn);
+    card.append(title, actions);
+
+    card.addEventListener("dragstart", (e) => {
+      e.dataTransfer.setData("text/plain", String(item.id));
+      card.classList.add("dragging");
+    });
+
+    card.addEventListener("dragend", () => card.classList.remove("dragging"));
+
+    return card;
+  }
+
+  function renderTasks(tasks) {
+    clearColumns();
+    tasks.forEach((t) => {
+      const card = createCard(t);
+      const status = t.status || "todo";
+      if (status === "todo") todoCol.appendChild(card);
+      else if (status === "inprogress") inprogressCol.appendChild(card);
+      else doneCol.appendChild(card);
+    });
+  }
+
+  async function fetchTasks() {
+    const res = await apiFetch(API_URL);
+    if (!res.ok) throw new Error("Failed to fetch");
+    currentTasks = await res.json();
+    renderTasks(currentTasks);
+    updateActionButtons();
+  }
+
+  async function createTask(title) {
+    const payload = { title, status: "todo" };
+    const res = await apiFetch(API_URL, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(payload),
+    });
+    if (!res.ok) throw new Error("Create failed");
+
+    const created = await res.json();
+    history.push({ type: "create", item: created });
+    saveStacks();
+    await fetchTasks();
```

Comment:
@Duong345 Những cái thao tác create, edit, delete chỉ 1 task mà em đang load lại toàn bộ task xong update lại render nhỉ, em thấy cái này có vấn đề gì ko ?